Objective-C/CBLDatabase.mm

```diff
@@ -1041,11 +1041,22 @@ - (void) scheduleDocumentExpiration: (NSTimeInterval)minimumDelay {
 
 - (void) purgeExpiredDocuments {
     dispatch_async(dispatch_get_global_queue( DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
-        CBLLog(Database, @"Purging expired documents...");
-        C4Error err;
-        UInt64 nPurged = c4db_purgeExpiredDocs(_c4db, &err);
-        CBLLog(Database, @"Purged %lld expired documents", nPurged);
-        [self scheduleDocumentExpiration: 1.0];
+        CBL_LOCK(self) {
+            if (!_c4db)
```

Comment:
technically the purgeExpiry shouldn't get called if the db doesn't exists(as we are cancelling all performSelectors while closing the DB).

Reply:
Same as my comment above. the actual purge is done in the background and could be schedule before cancelling the performSelectors.

Reply:
Scheduling is done on the main thread-run loop, and only the execution in background thread. So if the cancelling is done, before execution, it should remove it. anyways, we can discuss this in person.

Reply:
The cancelling is not relevant here as this is the case when the block to purge has already been scheduled on the background dispatch queue at the same time as the database is closed. Yes, we can discuss this later today.